FreeScape.Engine/Providers/MapProvider.cs

```diff
@@ -111,6 +113,7 @@ private void CacheTileSet(string name, TileSet tileSet)
                 cachedTileSetTiles.Add(tile.Id, cachedTileSetTile);
                 if (cachedTileSetTile.Animation is not null)
                 {
+                    _textureProvider.AddTexture(cachedTileSetTile.ImageTexture, $"animation:{cachedTileSetTile.Type}");
```

Comment:
I am not a fan of how this is done. We need to rework the texture provider to work with the Tiled texture stuff. instead of building and caching tiles and textures in the map provider